docs/madr/decisions/075-migrating-to-kri-based-envoy-resource-and-stat-naming.md

```diff
@@ -0,0 +1,193 @@
+# Migrating to KRI-based Envoy resource and stat naming
+
+* Status: accepted
+
+## Context and problem statement
+
+As described in the [Resource Identifier MADR](070-resource-identifier.md), we chose to standardize resource naming using the KRI (Kuma Resource Identifier) format. This change improves consistency across the system and allows for stronger cross-resource references, making data easier to present and understand. However, moving to KRI will break existing setups that rely on current Envoy stat and resource names, such as dashboards and GUI features. A smooth migration path is needed to let users adopt KRI without disruption.
+
+### Affected systems and users
+
+Two main groups are impacted:
+
+* Internal systems: Kuma GUI
+* External users: Dashboards, Prometheus
+
+#### Kuma GUI
+
+The Dataplane view displays inbound and outbound endpoints using data from Envoy stats. It matches these stats to configuration by parsing stat names, such as cluster or listener names. Changes to stat naming or resource identifiers will break this matching logic. We must coordinate changes with the GUI to support both the old and new formats during the transition.
+
+## Scope
+
+This decision applies to environments using the new service discovery model based on:
+
+* `MeshService`
+* `MeshExternalService`
+* `MeshMultiZoneService`
+
+These and the legacy `kuma.io/service` tag represent two different modes of describing the same types of services in the mesh. Since `kuma.io/service` is deprecated and will be removed in the future, this migration only covers naming changes for resources generated from the new model. Updating or supporting naming for the legacy mode is out of scope.
+
+This document also does not cover renaming of internal Envoy resources that do not correspond directly to real Kuma resources such as `MeshHTTPRoute`, `Dataplane`, or `MeshExternalService`. These include:
+
+* Secrets
```

Comment:
Is the secret internal object? We are going to introduce a new resource `MeshIdentity` and maybe we need to name  secret based on this.

Reply:
secrets are covered by the https://github.com/kumahq/kuma/pull/13783